sdk/python/tests/expediagroup/test_milvus_online_store.py

```diff
@@ -431,3 +455,81 @@ def test_milvus_online_write_batch(self, repo_config, caplog):
             assert progress["total_rows"] == total_rows_to_write
+
+    def test_milvus_online_read(self, repo_config, caplog):
+
+        # Generating data to directly load into Milvus
+        data_to_load = [
+            [i for i in range(10)],
+            [i + 2000 for i in range(10)],
+            [[random.random() for _ in range(2)] for _ in range(10)],
+        ]
+
+        # Creating a common schema for collection to directly add to Milvus
+        schema = CollectionSchema(
+            [
+                FieldSchema("film_id", DataType.INT64, is_primary=True),
+                FieldSchema("film_date", DataType.INT64),
+                FieldSchema("films", dtype=DataType.FLOAT_VECTOR, dim=2),
+            ]
+        )
+
+        # Creating an end-to-end example with just using pymilvus to understand how it works
+        with MilvusConnectionManager(repo_config.online_store):
+            # Create a collection
+            collection = Collection(name=self.collection_to_write, schema=schema)
+            # Drop all indexes if any exists
+            collection.drop_index()
+            # Create a new index
+            index_param = {"index_type": "FLAT", "metric_type": "L2", "params": {}}
+            collection.create_index("films", index_param)
+            collection.insert(data_to_load)
+            collection.load()
+
+        feast_schema = [
+            Field(
+                name="film_id",
+                dtype=Int64,
+                tags={"is_primary": "True", "description": "int64"},
+            ),
+            Field(
+                name="film_date",
+                dtype=Int64,
+                tags={
+                    "is_primary": "False",
+                    "description": "Int64",
+                },
+            ),
+            Field(
+                name="films",
+                dtype=Array(Float32),
+                tags={
+                    "is_primary": "False",
+                    "description": "float32",
+                    "dimensions": 2,
+                    "index_type": IndexType.hnsw.value,
+                },
+            ),
+        ]
+
+        film_data_with_entity_keys = (
+            self.create_n_customer_test_samples_milvus_online_read()
+        )
+        entity_keys, features, *rest = zip(*film_data_with_entity_keys)
+        feature_list = ["film_date", "films", "film_id"]
+
+        result = MilvusOnlineStore().online_read(
+            config=repo_config,
+            table=FeatureView(
+                name=self.collection_to_write, schema=feast_schema, source=SOURCE
+            ),
+            entity_keys=entity_keys,
+            requested_features=feature_list,
+        )
+
+        assert result is not None
+        assert len(result) == 10
+        assert result[0]["film_id"].int64_val == 0
+        assert result[0]["film_date"].int64_val == 2000
+        assert result[-1]["film_id"].int64_val == 9
```

Comment:
how come this is at index -1?

Reply:
Just checking the first and last indexes

Reply:
result[-1] is checking the last item in the list of results